src/OrchardCore.Modules/OrchardCore.Lists/Services/ContainerService.cs

```diff
@@ -211,19 +219,22 @@ public async Task<IEnumerable<ContentItem>> QueryContainedItemsAsync(string cont
                     query = _session.Query<ContentItem>()
                         .With<ContainedPartIndex>(CreateOrderedContainedPartIndexFilter(null, afterValue, contentItemId))
                         .OrderBy(x => x.Order)
-                        .With<ContentItemIndex>(CreateOrderedContentIndexFilter(publishedOnly))
+                        .With<ContentItemIndex>(CreateOrderedContentIndexFilter(ContentsStatus.Published))
                         .Take(pager.PageSize + 1);
                 }
                 else
                 {
                     var afterValue = new DateTime(long.Parse(pager.After));
                     query = _session.Query<ContentItem>()
                         .With<ContainedPartIndex>(CreateOrderedContainedPartIndexFilter(null, null, contentItemId))
-                        .With<ContentItemIndex>(CreateDefaultContentIndexFilter(null, afterValue, publishedOnly))
+                        .With<ContentItemIndex>(CreateDefaultContentIndexFilter(null, afterValue, containeditemOptions.Status))
                         .OrderByDescending(x => x.CreatedUtc)
                         .Take(pager.PageSize + 1);
                 }
 
+                if (!string.IsNullOrEmpty(containeditemOptions?.DisplayText) || containeditemOptions.Status != (int)ContentsStatus.None)
```

Comment:
Ahelyett, hogy ez a check háromszor szerepel, lehetne belül a `ContentItemFilter()`-ben.

Reply:
Erre azért gondoltam, hogy itt lehet szükség, mert az if-ben lévő utasítás hogy a functionban lévő return értéket csak akkor tegyük bele a query változóba, ha van szűrés. Mikor nem volt ott az if, akkor minden esetben lefutatta a szűréshez tartozó ellenőrzéseket.

Reply:
Igen, de ez az `if` lehetne a metódusban is, semmi szükség arra, hogy háromszor legyen lemásolva. Úgy nem fog az ottani szűrés lefutni, ha rögtön az elején return-ölsz ilyen esetben.